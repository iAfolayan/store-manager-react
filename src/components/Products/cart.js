import React, { Component } from 'react';
import { connect } from 'react-redux';
import { removeFromCart } from '../../actions/productsActions';

export class cart extends Component {
	// calculateTotal() {
	// 	const total = cartItems.reduce(
	// 		(curr, next) => {
	// 			curr.quantity += next.quantity;
	// 			curr.amount += next.price * next.quantity;
	// 			return curr;
	// 		},
	// 		{ amount: 0, quantity: 0 }
	// 	);
	// }

	remove(cartIndex) {
		this.props.remove(cartIndex);
	}

	showInfo(cart) {
		if (cart.length) return '';
		else return 'hide';
	}

	render() {
		const { cart } = this.props.products;
		return (
			<div id='cart' className={`cart ${this.showInfo(cart)}`}>
				<span className='cartHeading'>Cart Info</span>
				<hr />
				{cart.map((each, index) => (
					<ul className='cartItemsList' key={index}>
						<img src={each.image} className='img-thumbnail' id='item-img' alt='' />
						<div className='item-text'>
							<p id='cart-item-title' className='cart-item-title'>
								{each.name}
							</p>
							<span>&#8358;</span>
							<span id='item-price' className='item-price'>
								{each.price}.00 -- {each.quantity}
							</span>
							<button onClick={() => this.remove(index)}>Remove</button>
						</div>
					</ul>
				))}
				{
					<div className='cartTotal'>
						Total: &#8358;<span id='amount'>0</span>
					</div>
				}
				<a href='#'>
					<button type='button' className='cartBtn'>
						Proceed to checkout
					</button>
				</a>
			</div>
		);
	}
}

const mapStateToProps = (state) => ({
	products: state.productsReducer
});

const mapDispatchToProps = (dispatch) => ({
	remove: (cartIndex) => dispatch(removeFromCart(cartIndex))
});

export default connect(mapStateToProps, mapDispatchToProps)(cart);
