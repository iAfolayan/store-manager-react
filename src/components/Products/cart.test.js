import React from 'react';
import { shallow, mount } from 'enzyme';
import cart from './cart';
import sinon from 'sinon';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';


const props = {
   cart: [{
    productname: 'Mobile',
    id: 1,
    image: 'mobile.png',
    description: 'This is lovely and interesting',
    price: 340,
    minimumallowed: 40
  }]
}

const mockStore = configureStore([thunk]);
const store = mockStore({ ...props });

describe('<Cart />', () => {
  test('should test cart info', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <cart {...props} />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});