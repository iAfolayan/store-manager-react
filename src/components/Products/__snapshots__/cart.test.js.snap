// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cart /> should test cart info 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeState": Object {
        "cart": Array [
          Object {
            "description": "This is lovely and interesting",
            "id": 1,
            "image": "mobile.png",
            "minimumallowed": 40,
            "price": 340,
            "productname": "Mobile",
          },
        ],
      },
    }
  }
>
  <cart
    cart={
      Array [
        Object {
          "description": "This is lovely and interesting",
          "id": 1,
          "image": "mobile.png",
          "minimumallowed": 40,
          "price": 340,
          "productname": "Mobile",
        },
      ]
    }
  />
</ContextProvider>
`;
